modules/standard/Random.chpl

```diff
@@ -230,15 +232,15 @@ module Random {
 
   pragma "no doc"
   /* Actual implementation of choice() */
-  proc _choice(stream, arr: [], size:?sizeType, replace, prob:?probType)
+  proc _choice(stream, X: domain, size:?sizeType, replace, prob:?probType)
     throws
   {
 
-    if arr.rank != 1 {
-      compilerError('choice() array must be 1 dimensional');
+    if X.rank != 1 {
+      compilerError('choice() domain must be 1 dimensional');
```

Comment:
Should these error messages be more generic, since `x` can be an array, domain, or range?

Maybe just use identifier `x`?

```suggestion
      compilerError('choice() argument x must be 1 dimensional');
```

Reply:
Yes, Agreed.